lib/scope.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+function Scope(arr) {
+  if (!(this instanceof Scope)) {
+    return new Scope(arr);
+  }
+  if (!arr) {
+    arr = [];
+  } else if (arr instanceof Scope) {
+    return arr;
+  } else if (typeof arr === 'string') {
+    arr = arr.split(' ');
+  }
+  var obj = {};
+  for (var i = 0; i < arr.length; i++) {
+    obj[arr[i]] = true;
+  }
+  this._values = obj;
+}
+
+Scope.prototype = {
+
+  _values: undefined,
+
+  has: function has(scope) {
+    var that = this;
```

Comment:
nit: I find `that` confusing as a name here, because it sounds like it's referring to some other object rather than the object in question; please consider `self` or `me` or similar.